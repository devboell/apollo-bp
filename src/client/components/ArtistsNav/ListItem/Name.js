import styled from 'styled-components';

const Name = styled.h3`
  color: rgb(222, 81, 11);
`;

export default Name;
