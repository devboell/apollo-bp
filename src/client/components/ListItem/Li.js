import styled from 'styled-components';

const Li = styled.li`
  color: rgb(222, 81, 11);
`;

export default Li;
