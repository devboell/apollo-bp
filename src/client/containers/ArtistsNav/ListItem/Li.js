import styled from 'styled-components'

const Li = styled.li`
  > a {
    text-decoration: none;
  }
`;

export default Li
