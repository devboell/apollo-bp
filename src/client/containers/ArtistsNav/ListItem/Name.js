import styled from 'styled-components'

const Name = styled.h3`
  display: inline-block;
  margin: 10px;
  border-bottom: ${props => (props.selected ? 'solid gray 1px' : 'solid transparent 1px')};
  color: rgb(222, 81, 11);
`

export default Name
