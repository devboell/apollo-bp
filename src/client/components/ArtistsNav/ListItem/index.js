import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import Name from './Name'

const ListItem = ({ name }) =>
  <li>
    <Link to={`/${name}`}>
      <Name>{name}</Name>
    </Link>
  </li>


ListItem.propTypes = {
  name: PropTypes.string.isRequired,
}


export default ListItem
