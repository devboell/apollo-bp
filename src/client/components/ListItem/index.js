import React, { PropTypes } from 'react'
import Li from './Li'

const ListItem = ({ name }) =>
  <Li><h3>{name}</h3></Li>


ListItem.propTypes = {
  name: PropTypes.string.isRequired
}


export default ListItem
