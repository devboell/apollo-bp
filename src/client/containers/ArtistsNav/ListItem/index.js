import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import Li from './Li'
import Name from './Name'

const ListItem = ({ name, selected, onSelect }) =>
  <Li>
    <Link
      to={`/${name}`}
      onClick={onSelect}
    >
      <Name {...{ selected }}>{name}</Name>
    </Link>
  </Li>


ListItem.propTypes = {
  name: PropTypes.string.isRequired,
  selected: PropTypes.bool.isRequired,
  onSelect: PropTypes.func.isRequired,
}


export default ListItem
