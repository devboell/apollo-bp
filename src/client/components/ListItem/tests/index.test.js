import React from 'react'
import renderer from 'react-test-renderer'
import ListItem from '../index'

it('ListItem component snapshot', () => {
  const tree = renderer.create(<ListItem name={'testName'}/>).toJSON()
  expect(tree).toMatchSnapshot()
});
