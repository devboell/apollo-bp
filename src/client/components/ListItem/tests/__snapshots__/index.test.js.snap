// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component snapshot 1`] = `
<li
  className="bfXghp"
>
  <h3>
    testName
  </h3>
</li>
`;
